src/array.cr

```diff
@@ -1770,4 +1770,12 @@ class Array(T)
       end
     end
   end
+
+  def bsearch
+    idx=(0...self.size).bsearch{|i|yield self[i]}
```

Comment:
I'd like `self.bsearch_index{|e|e}`, but is this block all right? In Ruby, we can use `(&Kernel.proc)`.